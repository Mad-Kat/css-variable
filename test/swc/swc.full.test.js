const transformFileSync = require("@swc/core").transformFileSync;
const path = require("path");
// const plugin = require("../../swc");

describe("production transform", () => {
    it("CSSVariable", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/CSSVariable.js"),
        {
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: false,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
    it("createVar", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/createVar.js"),
        {
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: false,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
    it("renamed", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/renamed.js"),
        {
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: false,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
  });

  describe("development transform", () => {
    it("CSSVariable", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/CSSVariable.js"),
        {
          envName: "development",
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: false,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
    it("createVar", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/createVar.js"),
        {
          envName: "development",
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: false,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
    it("renamed", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/renamed.js"),
        {
          envName: "development",
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: false,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
  });

  describe("production transform with displayName", () => {
    it("CSSVariable", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/CSSVariable.js"),
        {
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: true,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
    it("createVar", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/createVar.js"),
        {
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: true,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
    it("renamed", () => {
      const { code } = transformFileSync(
        path.join(__dirname, "../fixtures/renamed.js"),
        {
          jsc: {
            experimental: {
              plugins: [
                [
                  require.resolve(
                    "../../swc/target/wasm32-wasi/debug/swc_plugin_css_variable.wasm"
                  ),
                  {
                    basePath: path.join(__dirname,"../../src/"),
                    displayName: true,
                  },
                ],
              ],
            },
          },
        }
      );
      expect(code).toMatchSnapshot();
    });
  });