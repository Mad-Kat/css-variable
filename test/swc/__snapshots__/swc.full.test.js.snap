// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`development transform CSSVariable 1`] = `
"import { createVar } from "css-variable";
var primary = createVar("GC_OuQ0");
var secondary = createVar("GC_OuQ1", {
    value: "#fff"
});
var allColors = [
    "c1",
    "c2",
    "c3"
];
var theme = {
    colors: {
        primary: createVar("GC_OuQ2")
    }
};
var generated = allColors.map(function(color) {
    return createVar("GC_OuQ3", color);
});
var generatedWithFallback = allColors.map(function(color) {
    return createVar("GC_OuQ4", color, {
        value: "#000"
    });
});
export var fileHashBase64 = "GC_OuQ";
export var vars = [
    {
        value: primary.toString(),
        withoutName: "var(--GC_OuQ0)",
        withName: "var(--primary--GC_OuQ0)"
    },
    {
        value: secondary.toString(),
        withoutName: "var(--GC_OuQ1, #fff)",
        withName: "var(--secondary--GC_OuQ1, #fff)"
    },
    {
        value: theme.colors.primary.toString(),
        withoutName: "var(--GC_OuQ2)",
        withName: "var(--primary--colors--theme--GC_OuQ2)"
    },
    {
        value: generated[0].toString(),
        withoutName: "var(--GC_OuQ3-c1)",
        withName: "var(--generated--GC_OuQ3-c1)"
    },
    {
        value: generated[1].toString(),
        withoutName: "var(--GC_OuQ3-c2)",
        withName: "var(--generated--GC_OuQ3-c2)"
    },
    {
        value: generated[2].toString(),
        withoutName: "var(--GC_OuQ3-c3)",
        withName: "var(--generated--GC_OuQ3-c3)"
    },
    {
        value: generatedWithFallback[0].toString(),
        withoutName: "var(--GC_OuQ4-c1, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c1, #000)"
    },
    {
        value: generatedWithFallback[1].toString(),
        withoutName: "var(--GC_OuQ4-c2, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c2, #000)"
    },
    {
        value: generatedWithFallback[2].toString(),
        withoutName: "var(--GC_OuQ4-c3, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c3, #000)"
    }
];
"
`;

exports[`development transform createVar 1`] = `
"import { createVar } from "css-variable";
"
`;

exports[`development transform renamed 1`] = `
"import { CSSVariable as CSSVar, createVar as createVariable } from "css-variable";
var primary = new CSSVar();
var secondary = createVariable("n8-W8g0");
"
`;

exports[`production transform CSSVariable 1`] = `
"import { createVar } from "css-variable";
var primary = createVar("GC_OuQ0");
var secondary = createVar("GC_OuQ1", {
    value: "#fff"
});
var allColors = [
    "c1",
    "c2",
    "c3"
];
var theme = {
    colors: {
        primary: createVar("GC_OuQ2")
    }
};
var generated = allColors.map(function(color) {
    return createVar("GC_OuQ3", color);
});
var generatedWithFallback = allColors.map(function(color) {
    return createVar("GC_OuQ4", color, {
        value: "#000"
    });
});
export var fileHashBase64 = "GC_OuQ";
export var vars = [
    {
        value: primary.toString(),
        withoutName: "var(--GC_OuQ0)",
        withName: "var(--primary--GC_OuQ0)"
    },
    {
        value: secondary.toString(),
        withoutName: "var(--GC_OuQ1, #fff)",
        withName: "var(--secondary--GC_OuQ1, #fff)"
    },
    {
        value: theme.colors.primary.toString(),
        withoutName: "var(--GC_OuQ2)",
        withName: "var(--primary--colors--theme--GC_OuQ2)"
    },
    {
        value: generated[0].toString(),
        withoutName: "var(--GC_OuQ3-c1)",
        withName: "var(--generated--GC_OuQ3-c1)"
    },
    {
        value: generated[1].toString(),
        withoutName: "var(--GC_OuQ3-c2)",
        withName: "var(--generated--GC_OuQ3-c2)"
    },
    {
        value: generated[2].toString(),
        withoutName: "var(--GC_OuQ3-c3)",
        withName: "var(--generated--GC_OuQ3-c3)"
    },
    {
        value: generatedWithFallback[0].toString(),
        withoutName: "var(--GC_OuQ4-c1, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c1, #000)"
    },
    {
        value: generatedWithFallback[1].toString(),
        withoutName: "var(--GC_OuQ4-c2, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c2, #000)"
    },
    {
        value: generatedWithFallback[2].toString(),
        withoutName: "var(--GC_OuQ4-c3, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c3, #000)"
    }
];
"
`;

exports[`production transform createVar 1`] = `
"import { createVar } from "css-variable";
"
`;

exports[`production transform renamed 1`] = `
"import { CSSVariable as CSSVar, createVar as createVariable } from "css-variable";
var primary = new CSSVar();
var secondary = createVariable("n8-W8g0");
"
`;

exports[`production transform with displayName CSSVariable 1`] = `
"import { createVar } from "css-variable";
var primary = createVar("primary--GC_OuQ0");
var secondary = createVar("secondary--GC_OuQ1", {
    value: "#fff"
});
var allColors = [
    "c1",
    "c2",
    "c3"
];
var theme = {
    colors: {
        primary: createVar("primary--colors--theme--GC_OuQ2")
    }
};
var generated = allColors.map(function(color) {
    return createVar("generated--GC_OuQ3", color);
});
var generatedWithFallback = allColors.map(function(color) {
    return createVar("generatedWithFallback--GC_OuQ4", color, {
        value: "#000"
    });
});
export var fileHashBase64 = "GC_OuQ";
export var vars = [
    {
        value: primary.toString(),
        withoutName: "var(--GC_OuQ0)",
        withName: "var(--primary--GC_OuQ0)"
    },
    {
        value: secondary.toString(),
        withoutName: "var(--GC_OuQ1, #fff)",
        withName: "var(--secondary--GC_OuQ1, #fff)"
    },
    {
        value: theme.colors.primary.toString(),
        withoutName: "var(--GC_OuQ2)",
        withName: "var(--primary--colors--theme--GC_OuQ2)"
    },
    {
        value: generated[0].toString(),
        withoutName: "var(--GC_OuQ3-c1)",
        withName: "var(--generated--GC_OuQ3-c1)"
    },
    {
        value: generated[1].toString(),
        withoutName: "var(--GC_OuQ3-c2)",
        withName: "var(--generated--GC_OuQ3-c2)"
    },
    {
        value: generated[2].toString(),
        withoutName: "var(--GC_OuQ3-c3)",
        withName: "var(--generated--GC_OuQ3-c3)"
    },
    {
        value: generatedWithFallback[0].toString(),
        withoutName: "var(--GC_OuQ4-c1, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c1, #000)"
    },
    {
        value: generatedWithFallback[1].toString(),
        withoutName: "var(--GC_OuQ4-c2, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c2, #000)"
    },
    {
        value: generatedWithFallback[2].toString(),
        withoutName: "var(--GC_OuQ4-c3, #000)",
        withName: "var(--generatedWithFallback--GC_OuQ4-c3, #000)"
    }
];
"
`;

exports[`production transform with displayName createVar 1`] = `
"import { createVar } from "css-variable";
"
`;

exports[`production transform with displayName renamed 1`] = `
"import { CSSVariable as CSSVar, createVar as createVariable } from "css-variable";
var primary = new CSSVar();
var secondary = createVariable("secondary--n8-W8g0");
"
`;
